packages/gatsby-source-wordpress/src/store.ts

```diff
@@ -1,14 +1,46 @@
-import { init } from "@rematch/core"
+import { RematchStore, init } from "@rematch/core"
 import immerPlugin from "@rematch/immer"
-import models from "./models"
-
-// import type { RematchStore } from "@rematch/core"
-// @todo any used to be RematchStore<typeof models> but this isn't exactly right..
-// need to revisit this later. newer versions of rematch sorted TS out but
-// there are a lot of breaking changes for us it seems
-const store: any = init({
-  models,
-  plugins: [immerPlugin()],
-})
+import models, { IRootModel } from "./models"
+
+import { AsyncLocalStorage } from "async_hooks"
+import { IPluginOptions } from "./models/gatsby-api"
+import { GatsbyNodeApiHelpers } from "./utils/gatsby-types"
+
+export interface IGatsbyApiHook {
+  (helpers: GatsbyNodeApiHelpers, pluginOptions: IPluginOptions): Promise<void>
+}
+
+export type Store = RematchStore<IRootModel, Record<string, never>>
+
+export const asyncLocalStorage = new AsyncLocalStorage<Store>()
+
+const STORE_MAP = new Map<string, Store>()
+
+/**
+ * Wraps the API hook with the async local storage context
+ */
+
+export const wrapApiHook =
+  (hook: IGatsbyApiHook): IGatsbyApiHook =>
+  async (helpers, pluginOptions) => {
+    const typePrefix = pluginOptions.schema?.typePrefix ?? ``
+    if (!STORE_MAP.has(typePrefix)) {
+      STORE_MAP.set(
+        typePrefix,
+        init({
+          models,
+          plugins: [immerPlugin<IRootModel>()],
+        })
+      )
+    }
+
+    const store = STORE_MAP.get(typePrefix)
```

Comment:
We get the instance of the store, keyed by the plugin's type prefix

Reply:
This is great, I love this implementation. async storage is perfect for this. 
Looking at this block now I'm wondering if we should throw an error if there are multiple instances that have the same prefix. What do you think? Having 2 instances with the same prefix will break most things in whichever of those instances runs first

Reply:
otherwise I think we could possibly get at the plugin instance ID and use that. In that case multiple instances with the same prefix will work and all their nodes will be merged

Reply:
That's a good idea. I think using the instance ID as the key means we could also check that there if there's a prefix collision